csrc/ir/utils.cpp

```diff
@@ -38,7 +38,9 @@ bool isSimpleTVSet(Expr* expr) {
 
   return ldst->opType() == LoadStoreOpType::Set &&
       // The hasRoot() check is to prevent picking up Set.Permute ops here
-      !out_tv->hasRoot();
+      !out_tv->hasRoot() &&
+      // The hasAllocation() check is to prevent picking up restriding ops
+      !out_tv->hasAllocation();
```

Comment:
This is for `remove_bcast_squeeze`, where we are treating `isSimpleTVSet` as a mere copy.

Reply:
without this, that optimization pass would alter output without considering the impact on `contiguity` and `allocation domain` change when changing output.